### DISCUSSION TITLE
Deploy custom zip

### DISCUSSION THREAD

**Discussion by themartto** [serverless/serverless#11668] [Q&A]
Can I deploy custom zip file (one that I create with zip command for example) with serverless deploy?

Currently when I try to provide my own .zip with --package file I'm getting the following error:

No serverless-state.json file found in the package path you provided.
 
 The reason I want to do this is because the following issue: https://github.com/softprops/serverless-rust/issues/81

**Comment by pgrzesik** [comment]
Hey @themartto. What is included in that zip file? Only code for your function(s)? If that's the case, you can use `package.artifact` both on service or on individual function level

**Reply by themartto** [reply]
The zip file includes my compiled **bootstrap** file and **myProgram** binary that I want to execute from my rust code. I'll try the artifact option and report back.

**Reply by themartto** [reply]
Bit of a late response but the artifact option is not working as well
